src/index.js

```diff
@@ -40,91 +32,108 @@ function multibase (nameOrCode, buf) {
  * @param {string|number} nameOrCode - The multibase name or code number.
  * @param {Buffer} buf - The data to be encoded.
  * @returns {Buffer}
- * @memberof Multibase
  */
 function encode (nameOrCode, buf) {
-  const base = getBase(nameOrCode)
-  const name = base.name
+  const enc = encoding(nameOrCode)
 
-  return multibase(name, Buffer.from(base.encode(buf)))
+  return Buffer.concat([enc.codeBuf, Buffer.from(enc.encode(buf))])
 }
 
 /**
  * Takes a buffer or string encoded with multibase header, decodes it and
  * returns the decoded buffer
  *
- * @param {Buffer|string} bufOrString
+ * @param {Buffer|string} data
  * @returns {Buffer}
- * @memberof Multibase
  *
  */
-function decode (bufOrString) {
-  if (Buffer.isBuffer(bufOrString)) {
-    bufOrString = bufOrString.toString()
+function decode (data) {
+  if (Buffer.isBuffer(data)) {
+    data = data.toString()
   }
-
-  const code = bufOrString.substring(0, 1)
-  bufOrString = bufOrString.substring(1, bufOrString.length)
-
-  if (typeof bufOrString === 'string') {
-    bufOrString = Buffer.from(bufOrString)
+  const enc = encoding(data[0])
+  if (enc) {
+    return Buffer.from(enc.decode(data.substring(1)))
   }
 
-  const base = getBase(code)
-  return Buffer.from(base.decode(bufOrString.toString()))
+  throw new Error('Unsupported encoding')
 }
 
 /**
  * Is the given data multibase encoded?
  *
- * @param {Buffer|string} bufOrString
+ * @param {Buffer|string} data
  * @returns {boolean}
- * @memberof Multibase
  */
-function isEncoded (bufOrString) {
-  if (Buffer.isBuffer(bufOrString)) {
-    bufOrString = bufOrString.toString()
+function isEncoded (data) {
+  if (Buffer.isBuffer(data)) {
+    data = data.toString()
   }
 
   // Ensure bufOrString is a string
-  if (Object.prototype.toString.call(bufOrString) !== '[object String]') {
+  if (Object.prototype.toString.call(data) !== '[object String]') {
     return false
   }
 
-  const code = bufOrString.substring(0, 1)
   try {
-    const base = getBase(code)
-    return base.name
+    const enc = encoding(data[0])
+    return enc.name
   } catch (err) {
     return false
   }
 }
 
 /**
+ * Validate encoded data
+ *
  * @param {string} name
  * @param {Buffer} buf
- * @private
  * @returns {undefined}
  */
 function validEncode (name, buf) {
-  const base = getBase(name)
-  base.decode(buf.toString())
+  const enc = encoding(name)
+  enc.decode(buf.toString())
 }
 
-function getBase (nameOrCode) {
-  let base
+/**
+ * Get the encoding by name or code
+ *
+ * @param {string} nameOrCode
+ * @returns {Base}
```

Comment:
Nitpick: I'm a fan of documenting error conditions, so have a `@throws` here would be cool (thought this would then also need to bubble up to all functions calling this one).